core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -665,6 +690,38 @@ protected void importOSM() {
             ghStorage.getProperties().put("datareader.data.date", f.format(reader.getDataDate()));
     }
 
+    private List<CustomArea> readCustomAreas() {
+        // todo: this is basically what we did before in GraphHopperManaged
+        ObjectMapper localObjectMapper = Jackson.newObjectMapper();
+        final Path bordersDirectory = Paths.get(customAreasLocation);
+        List<JsonFeatureCollection> jsonFeatureCollections = new ArrayList<>();
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(bordersDirectory, "*.{geojson,json}")) {
+            for (Path borderFile : stream) {
+                try (BufferedReader reader = Files.newBufferedReader(borderFile, StandardCharsets.UTF_8)) {
+                    JsonFeatureCollection jsonFeatureCollection = localObjectMapper.readValue(reader, JsonFeatureCollection.class);
+                    jsonFeatureCollections.add(jsonFeatureCollection);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
`UncheckedIOException` ?

Reply:
I just took this from `GraphHopperManaged`, but yes I agree `UncheckedIOException` seems to be what we should use (since Java 8).